docs/examples.rst

```diff
@@ -0,0 +1,215 @@
+`DBus services <https://github.com/rhinstaller/anaconda/tree/master/pyanaconda/modules>`_ of
+the Anaconda Installer for more inspiration.
+
+Basic usage
+-----------
+
+Show the current hostname.
+
+.. code-block:: python
+
+    from dasbus.connection import SystemMessageBus
+    bus = SystemMessageBus()
+
+    proxy = bus.get_proxy(
+        "org.freedesktop.hostname1",
+        "/org/freedesktop/hostname1"
+    )
+
+    print(proxy.Hostname)
+
+Send a notification to the notification server.
+
+.. code-block:: python
+
+    from dasbus.connection import SessionMessageBus
+    bus = SessionMessageBus()
+
+    proxy = bus.get_proxy(
+        "org.freedesktop.Notifications",
+        "/org/freedesktop/Notifications"
+    )
+
+    id = proxy.Notify(
+        "", 0, "face-smile", "Hello World!",
+        "This notification can be ignored.",
+        [], {}, 0
+    )
+
+    print("The notification {} was send.".format(id))
+
+Handle a closed notification.
+
+.. code-block:: python
+
+    from dasbus.loop import EventLoop
+    loop = EventLoop()
+
+    from dasbus.connection import SessionMessageBus
+    bus = SessionMessageBus()
+
+    proxy = bus.get_proxy(
+        "org.freedesktop.Notifications",
+        "/org/freedesktop/Notifications"
+    )
+
+    def callback(id, reason):
+        print("The notification {} was closed.".format(id))
+
+    proxy.NotificationClosed.connect(callback)
+    loop.run()
+
+Run the service org.example.HelloWorld.
+
+.. code-block:: python
+
+    from dasbus.loop import EventLoop
+    loop = EventLoop()
+
+    from dasbus.connection import SessionMessageBus
+    bus = SessionMessageBus()
+
+    class HelloWorld(object):
+        __dbus_xml__ = """
+        <node>
+            <interface name="org.example.HelloWorld">
+                <method name="Hello">
+                    <arg direction="in" name="name" type="s" />
+                    <arg direction="out" name="return" type="s" />
+                </method>
+            </interface>
+        </node>
+        """
+
+        def Hello(self, name):
+            return "Hello {}!".format(name)
+
+    bus.publish_object("/org/example/HelloWorld", HelloWorld())
+    bus.register_service("org.example.HelloWorld")
+    loop.run()
+
+
+Features
+--------
+
+Use constants to define DBus services and objects.
+
+.. code-block:: python
+
+    from dasbus.connection import SystemMessageBus
+    from dasbus.identifier import DBusServiceIdentifier
+
+    NETWORK_MANAGER = DBusServiceIdentifier(
+        namespace=("org", "freedesktop", "NetworkManager"),
+        message_bus=SystemMessageBus()
+    )
+
+    proxy = NETWORK_MANAGER.get_proxy()
+    print(proxy.NetworkingEnabled)
+
+Use exceptions to propagate and handle DBus errors.
+
+.. code-block:: python
+
+    from dasbus.error import dbus_error, DBusError
+
+    @dbus_error("org.freedesktop.DBus.Error.InvalidArgs")
+    class InvalidArgs(DBusError):
+        pass
```

Comment:
I'm not sure I understand this one. Is it an exception that is automatically raised when DBus gives you an error in the namespace?

Reply:
Sorry for ambiguity, what I meant is - the description for this example does not help me understand what this is going to do. It's literally just a declaration.

Reply:
Fixed in https://github.com/rhinstaller/dasbus/pull/29.